Проверка заданий по HTML и CSS of Yuriy Bespechanskiy

Yuriy2437 opened the pull request:
В проекте 4 папки: cv_html - это визитка по HTML:
                                html_practice - 6 практических заданий по HTML, ссылка на них в файле README.
                                cv_css - это визитка по CSS
                                css_practice - 6 практических заданий по CSS, ссылка на них в файле README.

Yuriy2437 committed:
Добавлены 4 папки с заданиями HTML и CSS

foxprogs commented on css_practice/ex1/styles.css:
```diff
@@ -0,0 +1,27 @@
+
+a:visited {
+  color: grey;
+}
```
нет пустой строки в конце файла. Надо настроить чтобы в вскоде она добавлялась автоматом.

foxprogs commented on html_practice/ex5.html:
```diff
@@ -0,0 +1,38 @@
+    </tbody>
+  </table>
+</body>
+</html>
```
нет пустой строки в конце файла. Надо настроить чтобы в вскоде она добавлялась автоматом.

foxprogs replied:
![изображение](https://github.com/tt-dev-23/HTML_CSS_Yuriy/assets/30548361/fa29fa48-6f32-48cd-997e-eae1d78e9e3b)

Yuriy2437 committed:
Add files via upload

В файле INDEX.TXT ссылки на результаты выполнения всех заданий проекта

Yuriy2437 committed:
Delete INDEX.TXT

Удален INDEX.TXT. Вместо него будет залит index.html

Yuriy2437 committed:
Файл ссылок на все задания проекта

Добавлен файл index.html с открывающимися ссылками на все задания проекта HTML и CSS

Yuriy2437 committed:
Файл картинок-результатов выполнения заданий

В результате выполнения файла index.html на экран выводятся не https-ссылки, а непосредственно картинки, таблицы и другие результаты выполнения заданий проекта.

Yuriy2437 committed:
Еще один файл визуальных итогов заданий

Файл, где иным образом, чем в index_picture.html, исполнена задача показать все результаты учебного проекта в одном визуальном ряде.

Yuriy2437 committed:
Delete index_image.html

Файл удаляется, поскольку результаты оказались идентичны index_picture.html

foxprogs commented on cv_html/index.html:
```diff
@@ -0,0 +1,125 @@
+          <h1>Юрий Беспечанский</h1>
+          <p>
+            <em
+              >Я профессиональный христианский философ<strong> и начинающий Frontend-разработчик</strong>
```
так не стоит переносить, тег лучше писать на одной строке, `<em>`

foxprogs commented on cv_html/reference.html:
```diff
@@ -0,0 +1,16 @@
+      <li>
+        <a href="http://baznica.info/2020/08/uchenie-mastera-ekxarta-i-ego-vliyanie-na-bogoslovie-martina-lyutera/">
+          Учение Мастера Экхарта И Его Влияние На Богословие Мартина Лютера</a
+        >
```
снова тег разбит на две строки

foxprogs commented on html_practice/ex4.html:
```diff
@@ -0,0 +1,29 @@
+  <body>
+    <p>Сейчас мы выполняем упражнение 4</p>
+    <p>Изображения разных форматов:</p>
+    <ul type="none">
+      <li>
```
картинки не надо в списке выдавать. Просто друг за другом изображения, между ними br

foxprogs commented on html_practice/ex4.html:
```diff
@@ -0,0 +1,29 @@
+        <img src="https://free-png.ru/wp-content/uploads/2020/07/logo-you-tube-2.png" />
+      </li>
+      <li>
+        <img src="https://timeweb.com/ru/community/article/ff/ff0770833a0026f1b62813408cf5bfb6.jpg" />
```
картинка же очень большая. Можно ей указать размер. Да и остальным тоже

foxprogs commented on html_practice/ex4.html:
```diff
@@ -0,0 +1,29 @@
+        <img src="https://thumbs.gfycat.com/LargeVigilantCardinal-size_restricted.gif" />
+      </li>
+      <li>
+        <img src="https://www.flaticon.com/free-icon/svg-file-format-variant_29495#" />
```
А что это? Это же не ссылка на картинку, она не открывается. В задании написано что надо отобразить картинку в формате svg.

foxprogs commented on html_practice/ex4.html:
```diff
@@ -0,0 +1,29 @@
+    </ul>
+    <p>Логотип YouTube:</p>
+    <p>
+      <img src="https://free-png.ru/wp-content/uploads/2020/07/logo-you-tube-2-1.png" alt="Логотип YouTube" />
```
в задании написано "Вставить изображение логотипа YouTube и реализовать переход на сайт YouTube по клику на картинку." Сейчас по клику на картинку мы никуда не переходим

foxprogs commented on html_practice/ex6.html:
```diff
@@ -0,0 +1,38 @@
+  </head>
+  <body>
+    <p>Привет, я риелтор! Свяжитесь со мной по интересующему вас вопросу.</p>
+    <form action="mailto:Byura@rambler.ru" method="post" enctype="text/plain">
```
не надо на почту. Можно тут просто написать `<form>` тогда он по умолчанию считает что надо отправить методом GET на тот же урл. И при отправке в адресной строке будет видно все что было заполнено в форме

foxprogs replied:
а, не досмотрел что про почту написано в условии

foxprogs commented on html_practice/ex6.html:
```diff
@@ -0,0 +1,38 @@
+        <label for="name">Имя</label>
+        <input type="text" id="name" name="name" />
+      </p>
+      <label for="date">Дата для встречи</label>
+      <input type="date" id="date" name="date" />
+      <p></p>
```
тут неправильно написано. надо `Для разделения содержимого страницы на разные строчки используйте тег <p>` а тут тег `<p>` сам по себе

foxprogs commented on index.html:
```diff
@@ -0,0 +1,88 @@
+    <p>
+      HTML-визитка
+      <a href="https://tt-dev-23.github.io/HTML_CSS_Yuriy/cv_html/index.html" target="_blank"
+        >https://tt-dev-23.github.io/HTML_CSS_Yuriy/cv_html/index.html
```
так не стоит переносить
можно к примеру так, каждый атрибут на новой строке
```html
<a
  href="https://tt-dev-23.github.io/HTML_CSS_Yuriy/cv_html/index.html"
  target="_blank">
```

foxprogs commented on index_picture.html:
это все же не picture а фреймы

foxprogs commented on cv_css/index.html:
```diff
@@ -0,0 +1,123 @@
+          <li>
+            <a href="http://baznica.info/2020/08/uchenie-mastera-ekxarta-i-ego-vliyanie-na-bogoslovie-martina-lyutera/"
+              >Учение Майстера Экхарта и его влияние на богословие Мартина Лютера</a
+            >
```
вот так тоже не надо. Надо везде в общем это исправить

foxprogs commented on css_practice/ex1/index.html:
```diff
@@ -0,0 +1,34 @@
+    <link href="styles.css" rel="stylesheet" />
+  </head>
+  <body>
+    <div class="text">
```
я уже писал, в исходных данных нет этого дива `<div class="text">`

foxprogs commented on css_practice/ex1/index.html:
```diff
@@ -0,0 +1,34 @@
+        <li>Размещение элементов на странице</li>
+        <li>
+          Адаптивность страниц -
+          <div class="italic">
```
этого дива тоже нет

foxprogs commented on css_practice/ex4/index.html:
```diff
@@ -0,0 +1,15 @@
+  </head>
+  <body>
+    <ul class="list-group">
+      <li>HTML</li>
```
должны быть классы `class="list-group-item"` для `li`

foxprogs commented on css_practice/ex5/index.html:
```diff
@@ -0,0 +1,15 @@
+  </head>
+  <body>
+    <ul class="list-group">
+      <li><a href="https://youtu.be/DOEtVdkKwcU">HTML</a></li>
```
тут так же не хватает классов у элемента списка

foxprogs commented on css_practice/ex5/index.html:
![изображение](https://github.com/tt-dev-23/HTML_CSS_Yuriy/assets/30548361/8d827e95-2332-4696-94e5-6cef60e216c7)
отличается от этого 
![изображение](https://github.com/tt-dev-23/HTML_CSS_Yuriy/assets/30548361/75a24a6a-84b0-400b-a831-f0d986cfe77a)

foxprogs commented on css_practice/ex6/styles.css:
```diff
@@ -0,0 +1,38 @@
+
+.card-img-top {
+  width: 100%;
+  border-top-left-radius: 10px;
+  border-top-right-radius: 10px;
```
так не очень, если увеличить то в уголках там есть пустота. Лучше для всей карточки добавить overflow: hidden; тогда все что выдохит за края карточки - будет просто скрываться

foxprogs commented on css_practice/ex6/styles.css:
```diff
@@ -0,0 +1,38 @@
+  background: #e45d80;
+  color: #fff;
+  padding: 10px 15px;
+  border: none;
+  font-family: inherit;
```
для чего эти два стиля?

foxprogs reviewed (changes requested):
нужно исправить

Yuriy2437 committed:
Сделаны изменения в соответствии с замечаниями

Yuriy2437 committed:
Update index.html

Вручную на GitHub изменены разрывы тэга <a>/</a> в строках 89-91, так как в VSCode Prettier не позволял это выполнить, автоматически форматируя код

foxprogs commented on html_practice/ex4.html:
```diff
@@ -0,0 +1,28 @@
+      height="300"
+    />
+    <br />
+    <img src="https://thumbs.gfycat.com/LargeVigilantCardinal-size_restricted.gif" width="500" height="300" />
```
высоту не надо указывать иначе картинки искажаются, только ширину оставить и все

Yuriy2437 committed:
Исправлено задание ex4.html
Теперь на картинке отображается файл johny.svg

Yuriy2437 committed:
Изменен код задания CSS_1 без классов

Yuriy2437 committed:
Убрал высоту картинок в задании HTML_4

Yuriy2437 committed:
Картинка CSS_5 теперь соответствует заданной

foxprogs commented on css_practice/ex1/styles.css:
```diff
@@ -0,0 +1,27 @@
+  font-weight: 700;
+}
+
+ul li:nth-child(5) > em {
```
А для чего такой сложный класс?
Можно просто `li em {`

Yuriy2437 committed:
Изменен код заданий CSS_4 и CSS_5

Yuriy2437 committed:
Изменены стили границ в CSS_5

Yuriy2437 committed:
Еще одно изменение границ в CSS_5

foxprogs commented on css_practice/ex4/styles.css:
```diff
@@ -0,0 +1,34 @@
+  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
+}
+
+.list-group li a {
+  color: black;
+  text-decoration: none;
+}
```
в 4 задании же нет ссылок, для чего этот стиль

foxprogs reviewed (commented):
пару исправлений еще

Yuriy2437 committed:
Последние поправки к заданиям CSS_1 и CSS_4

foxprogs reviewed (approved):
можно мерджить
